controllers/masteruserrecord/masteruserrecord_controller_test.go

```diff
@@ -78,10 +84,81 @@ func TestAddFinalizer(t *testing.T) {
 			})
 	})
 
-	t.Run("fails because it cannot add finalizer", func(t *testing.T) {
+	t.Run("no condition set when there is no SpaceBinding/Space", func(t *testing.T) {
 		// given
-		mur := murtest.NewMasterUserRecord(t, "john")
+		memberClient := commontest.NewFakeClient(t)
 		hostClient := commontest.NewFakeClient(t, mur)
+		InitializeCounters(t, NewToolchainStatus(
+			WithMember(commontest.MemberClusterName),
+			WithMetric(toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey, toolchainv1alpha1.Metric{
+				string(metrics.Internal): 1,
+			}),
+			WithMetric(toolchainv1alpha1.UserSignupsPerActivationAndDomainMetricKey, toolchainv1alpha1.Metric{
+				"1,internal": 1,
+			}),
+		))
+
+		cntrl := newController(hostClient, s, ClusterClient(commontest.MemberClusterName, memberClient))
+
+		// when
+		result, err := cntrl.Reconcile(context.TODO(), newMurRequest(mur))
+
+		// then
+		require.NoError(t, err)
+		require.False(t, result.Requeue)
+
+		murtest.AssertThatMasterUserRecord(t, "john", hostClient).
+			HasConditions().
```

Comment:
maybe minor thing - but should we set a condition to be able to easily identify that is an expected case and not an issue with our operators ?

Reply:
Good point. This is a bit tricky. There are two situations:
1. user is provisioned, space is supposed to be created, and we should wait until the UserAccount is present so we can mark the provisioning part done
2. user was already provisioned before, but all UserAccounts were removed from the MUR. This is a valid use-case (although it's not being used anywhere, at least for now). So the MUR had status UserAccounts before, but all of them were removed, but the MUR should be marked as "ready". This is exactly why I added the "inner `if` statement" in the `alignReadiness` function:
https://github.com/codeready-toolchain/host-operator/blob/7f8b407cb9ff0a80ee2464db47897f2428b01daf/controllers/masteruserrecord/sync.go#L197-L205

now imagine that we would set the "Ready" condition to "Provisioning" when we wait for the UserAccounts and the MUR gets reconciled for any reason again - how to distinguish between these two situations I described above?

Reply:
hmmm.. I maybe know how - let me try something

Reply:
this should do the job: [739aeb9](https://github.com/codeready-toolchain/host-operator/pull/872/commits/739aeb967b007a7ab4c7eaff24c94f61fb863729)